zeeguu/core/util/text.py

```diff
@@ -6,19 +6,247 @@
 from collections import Counter
 from nltk import SnowballStemmer
 from zeeguu.core.model.language import Language
+from logging import log
+from string import punctuation
+import re
+
 
 AVERAGE_SYLLABLE_LENGTH = 2.5
 
 """
     Collection of simple text processing functions
 """
+NLTK_SUPPORTED_LANGUAGES = set(
+    [
+        "czech",
+        "danish",
+        "dutch",
+        "english",
+        "estonian",
+        "finnish",
+        "french",
+        "german",
+        "greek",
+        "italian",
+        "norwegian",
+        "polish",
+        "portuguese",
+        "russian",
+        "slovene",
+        "spanish",
+        "swedish",
+        "turkish",
+    ]
+)
+
+
+class Token:
+    PUNCTUATION = "»«" + punctuation + "–—“‘”“’„¿"
+    LEFT_PUNCTUATION = "«<({#„¿["
```

Comment:
![image](https://github.com/user-attachments/assets/c643d712-49b4-4e5f-b7b5-bfc878029532)

in english the shape of left / right position aware double quotations is at the top: the 66 and 99 shapes. from: https://en.wikipedia.org/wiki/Quotation_mark#Summary_table